sklearn/cluster/optics_.py

```diff
@@ -278,16 +278,14 @@ def fit(self, X, y=None):
 
 
 def _validate_size(size, n_samples, param_name):
```

Comment:
we don't need to modify this function, a small PR is often easier to review.

Reply:
Actually that the  thing that changed... when validating the value its scaled to be in an appropriate region

Reply:
```
def _validate_size(size, n_samples, param_name):
    if isinstance(size, int) and size > 1 and size < n_samples:
        return size
    elif size <= 1 and size > 0:
        return int(max(2, size * n_samples))
    else:
        raise ValueError('%s must be a positive integer smaller than'
                         ' the number of samples or a float between 0 and 1. '
                         'Got %r' % (param_name, size))

```

Reply:
Maybe simply change ``max(2, min_samples * n_samples)`` to ``max(2, int(min_samples * n_samples))``?